File: bioskopina_mobile/lib/models/comment.dart
import 'package:json_annotation/json_annotation.dart';
part 'comment.g.dart';

@JsonSerializable()
class Comment {
  int? id;
  int? postId;
  int? userId;
  String? content;
  int? likesCount;
  int? dislikesCount;
  DateTime? dateCommented;

  Comment(
    this.id,
    this.postId,
    this.userId,
    this.content,
    this.likesCount,
    this.dislikesCount,
    this.dateCommented,
  );

  factory Comment.fromJson(Map<String, dynamic> json) =>
      _$CommentFromJson(json);

  Map<String, dynamic> toJson() => _$CommentToJson(this);
}

File: backend/Bioskopina.Services/Database/Comment.cs
using System;
using System.Collections.Generic;

namespace Bioskopina.Services.Database;

public partial class Comment
{
    public int Id { get; set; }

    public int PostId { get; set; }

    public User User { get; set; }
    
    public int UserId { get; set; }

    public string Content { get; set; } = null!;

    public int LikesCount { get; set; }

    public int DislikesCount { get; set; }

    public DateTime DateCommented { get; set; }

    public virtual Post Post { get; set; } = null!;


    public virtual ICollection<UserCommentAction> UserCommentActions { get; set; } = new List<UserCommentAction>();

}

